project-addons/purchase_picking/stock.py

```diff
@@ -52,7 +81,7 @@ def _get_picking_ids(self):
         self.picking_ids = res
 
     name = fields.Char("Container Ref.", required=True)
-    date_expected = fields.Date("Date expected", compute='_get_date_expected', readonly=True, required=False)
+    date_expected = fields.Date("Date expected", compute='_get_date_expected', readonly=False, required=False)
```

Comment:
¿por qué este cambio? al no tener función "set" no tiene sentido

Reply:
Porque queremos que se pueda cambiar la fecha del contenedor y eso afecte a la fecha programada de los albaranes

Reply:
¿Pero así tampoco se puede No? da la falsa sensación de que sí, porque el campo aparece editable, pero lo que pongas ahí no va a ningún sitio, excepto que definas una función set con "fnct_inv"

Reply:
Lo que pongo ahi, se modifica al entrar en el write que he definido después y entonces modifica la fecha de los albaranes. No se si está bien hecho así

Reply:
Podría valer, lo normal sería hacer algo muy similar pero definiendo una función en "fnct_inv", si lo dejas así controla en el write que realmente venga, date_expected en vals sino fallaría.

Reply:
Entonces le pongo una opción "fnct_inv" con una función que se llama dentro del método write para que inserte el valor que tiene el campo en BBDD no?

Reply:
Vale no, es al revés. En la función "fnct_inv" llamo al método write por lo que estoy viendo

Reply:
He hecho pruebas pero cuando pongo "fnct_inv" y una función,  el sistema no entra en ella ni al modificar el campo ni al guardar el objeto

Reply:
Sí, exacto

Reply:
Tiene que entrar, es cómo se hace, mira algún ejemplo en el estándar.

Reply:
No, si uso "fnct_inv" no entra porque eso solo funciona en api v7. En api v8 es "inverse" pero si uso eso no puedo hacer un write en la misma vista porque me entra en un bucle infinito de llamadas (el write llama a la función inversa y la función inversa al write). Si quito el write de la vista, no me vienen los datos que se modifican de la misma, esto es lo peor porque no tengo forma de mandarle a la función lo que yo quiero ya que me entra en otro write diferente.

Reply:
Ah, en la api que usabas no me fijara, pues inverse entonces. El write que tienes heredado tienes que quitarlo sino estás haciendo lo mismo dos veces, sólo debes hacer el write en el inverse y ya estaría.

Reply:
El problema es que en la funcion "inverse" no me llegan los valores de la vista que se han cambiado ¿Los tomo directamente desde el objeto?

Reply:
Sí, en self si es api.one o como le llames al objeto si es un api.multi dentro de un for, ahí verás el valor que metió el usuario y luego ya lo utilizas.